src/Traits/OrderItemCustomerOptionCapableTrait.php

```diff
@@ -111,7 +111,15 @@ public function equals(SyliusOrderItemInterface $item): bool
 
         // iterate over all options and compare their values
         foreach ($itemCustomerConfiguration as $code => $value) {
-            if ($curItemCustomerConfiguration[$code]->getOptionValue() !== $value->getOptionValue()) {
+            if (is_array($curItemCustomerConfiguration[$code]) && is_array($value)) {
+                $curItemCustomerConfigurationOptionValue = current($curItemCustomerConfiguration[$code])->getOptionValue();
+                $valueOptionValue                        = current($value)->getOptionValue();
+            } else {
+                $curItemCustomerConfigurationOptionValue = $curItemCustomerConfiguration[$code]->getOptionValue();
+                $valueOptionValue                        = $value->getOptionValue();
+            }
```

Comment:
Hmm faudrait pas gérer individuellement la condition ? Je sais pas si c'est possible mais là si `$curItemCustomerConfiguration[$code]` est un array et pas `$value`, ça va encore péter
Genre plutôt
```suggestion
            $curItemCustomerConfigurationOptionValue = (is_array($curItemCustomerConfiguration[$code]) ? current($curItemCustomerConfiguration[$code]) : $curItemCustomerConfiguration[$code])->getOptionValue();
            $valueOptionValue                        = (is_array($value) ? current($value) : $value)->getOptionValue();
```

Reply:
ha oui si carrement, j'etait parti sur un autre truc a la base, j'ai modifié pour ca, mais oui en effet separer encore ca serait mieux. En principe c'est les deux du meme type mais on sais jamais